viamonvif/discovery.go

```diff
@@ -109,6 +110,17 @@ type CameraInfo struct {
 	HardwareID      string   `json:"hardware_id"`
 }
 
+// regex to remove non alpha numerics.
+var reg = regexp.MustCompile("[^a-zA-Z0-9]+")
+
+// Name creates generates a name for the camera based on discovered information about the camera.
+func (cam *CameraInfo) Name() string {
+	stripManufacturer := reg.ReplaceAllString(cam.Manufacturer, "")
+	stripModel := reg.ReplaceAllString(cam.Model, "")
+	stripSerial := reg.ReplaceAllString(cam.SerialNumber, "")
+	return fmt.Sprintf("%s-%s-%s", stripManufacturer, stripModel, stripSerial)
+}
+
```

Comment:
@nicksanford here is the part that you may be concerned about - this gives us the same name per model manufacturer and serial

Reply:
I can add a number as a part of the input if people want to make the names different. 

Personally I think its okay if the same camera has the same name for different streams to help prevent users from configuring a camera multiple times.

Reply:
Hm.. But a user may want to configure multiple streams for a given "camera". E.g. NVR/XVR with multiple channels

Reply:
I'll go ahead and update the function to take in a channel number if we think thats a normal use case. Other option would be to add the channel to `CameraInfo` but I'm not familiar enough with the code to know where to find that

`fmt.Sprintf("%s-%s-%s-channel%v", stripManufacturer, stripModel, stripSerial, channel)`